iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ValidationUtil.java

```diff
@@ -383,13 +518,302 @@ public static void checkPhaseTapChangerRegulation(Validable validable, PhaseTapC
+        }
+    }
+
+    /**
+     * @deprecated
+     * Use {@link #checkVoltageSetpoint(Validable, String, double, boolean)} instead
+     */
+    @Deprecated(since = "4.4.0")
+    public static boolean checkVoltageControl(Validable validable, Boolean voltageRegulatorOn, double voltageSetpoint) {
+        if (voltageRegulatorOn == null) {
+            throw new ValidationException(validable, "voltage regulator status is not set");
+        }
+        boolean bVoltageRegulatorOn = voltageRegulatorOn; // make sonar happy java:S5411 Boxed "Boolean" should be avoided in boolean expressions
+        if (bVoltageRegulatorOn) {
+            if (Double.isNaN(voltageSetpoint) || voltageSetpoint <= 0) {
+                throw createInvalidValueException(validable, voltageSetpoint, VOLTAGE_SETPOINT, "voltage regulator is on");
+            }
+            return false;
+        }
+        return true;
+    }
+
+    public static boolean validRegulatingVoltageControl(Terminal regulatingTerminal, double voltageSetpoint, Network network) {
+        return validRegulatingTerminal(regulatingTerminal, network) && validVoltageSetpoint(voltageSetpoint);
+    }
+
+    public static boolean validRegulatingVoltageControl(Terminal regulatingTerminal, double voltageSetpoint, double targetDeadband, Network network) {
+        return validRegulatingTerminal(regulatingTerminal, network) && validVoltageSetpoint(voltageSetpoint) && validDeadband(targetDeadband);
+    }
+
+    public static boolean validRegulatingVoltageControl(double voltageSetpoint) {
+        return validVoltageSetpoint(voltageSetpoint);
+    }
+
+    public static boolean validRegulatingReactivePowerControl(Terminal regulatingTerminal, double reactivePowerSetpoint, Network network) {
+        return validRegulatingTerminal(regulatingTerminal, network) && validReactivePowerSetpoint(reactivePowerSetpoint);
+    }
+
+    public static boolean validRegulatingCurrentOrActivePowerControl(Terminal regulatingTerminal, double valueSetpoint, double targetDeadband, Network network) {
+        return validRegulatingTerminal(regulatingTerminal, network) && validCurrentOrActivePowerSetpoint(valueSetpoint) && validDeadband(targetDeadband);
+    }
+
+    private static boolean validRegulatingTerminal(Terminal regulatingTerminal, Network network) {
+        return regulatingTerminal != null && regulatingTerminal.getVoltageLevel().getNetwork() == network;
+    }
+
+    private static boolean validVoltageSetpoint(double voltageSetpoint) {
+        return !Double.isNaN(voltageSetpoint) && voltageSetpoint > 0;
+    }
+
+    private static boolean validReactivePowerSetpoint(double reactivePowerSetpoint) {
+        return !Double.isNaN(reactivePowerSetpoint);
+    }
+
+    private static boolean validCurrentOrActivePowerSetpoint(double valueSetpoint) {
+        return !Double.isNaN(valueSetpoint);
+    }
+
+    private static boolean validDeadband(double deadband) {
+        return !Double.isNaN(deadband) && deadband >= 0;
```

Comment:
It is a bit weird because you do not check if values are not NaN, which should be checked when regulating

Reply:
I do not quite understand what you mean. Could you please clarify this point?